account/src/main/java/com/caldevsupplychain/account/controller/AccountControllerImpl.java

```diff
@@ -84,24 +86,161 @@
 		}
+		return new ResponseEntity<>(userMapper.toWS(user), HttpStatus.CREATED);
+	}
+
+	@GetMapping("/activate/{token}")
+	public ResponseEntity<?> activateAccount(@PathVariable("token") String token) {
+		Optional<UserBean> user = accountService.findByToken(token);
+
+		if (!user.isPresent()) {
+			return new ResponseEntity<Object>(new ApiErrorsWS(ErrorCode.INVALID_TOKEN.name(), "Invalid Token."), HttpStatus.BAD_REQUEST);
+		}
+		accountService.activateUser(user.get().getId());
 
-		return new ResponseEntity<>(userMapper.userBeanToWS(user), HttpStatus.CREATED);
+		// TODO: need to tell FE to redirect to /login page after activate signup token
+		return new ResponseEntity<>(userMapper.toWS(user.get()), HttpStatus.OK);
 	}
 
+	@PostMapping("/login")
+	public ResponseEntity<?> login(@Validated @RequestBody UserWS userWS) {
+
+		Optional<UserBean> user = accountService.findByEmailAddress(userWS.getEmailAddress());
+
+		if (!user.isPresent()) {
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGIN_INVALID.name(), "Invalid Login"), HttpStatus.UNAUTHORIZED);
+		}
+
+		Subject subject = SecurityUtils.getSubject();
+
+		// Apache Shiro authentication check for 1st login
+		if (!subject.isAuthenticated()) {
+			UsernamePasswordToken token = new UsernamePasswordToken(userWS.getEmailAddress(), userWS.getPassword());
+
+			try {
+				subject.login(token);
+			} catch (AuthenticationException e) {
+				log.error("Error in login. User fail to login. user={}", user.toString());
+				return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGIN_INVALID.name(), e.getMessage()), HttpStatus.UNAUTHORIZED);
+			}
+		}
+
+		log.info("Success in user login. user={}", user.toString());
+
+		UserBean userBean = user.get();
+		JwtBean jwtBean = jwtMapper.toBean(userBean);
+		JWTAuthenticationToken jwtObj = jwtService.createJwtToken(jwtBean);
+
+		log.info("Success create jwt token. jwtToken={}", jwtObj.getToken());
+
+		return new ResponseEntity<>(userMapper.toWS(userBean), jwtService.createJwtHeader(jwtObj.getToken()), HttpStatus.OK);
+	}
+
+
+	@GetMapping("/logout")
+	@RequiresAuthentication
+	public ResponseEntity<?> logout() {
+		Subject subject = SecurityUtils.getSubject();
+
+		Optional<UserBean> user = accountService.findByUuid(subject.getPrincipal().toString());
+
+		if (!user.isPresent()) {
+			log.error("Error in logout. User fail to logout. user={}", user.toString());
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGOUT_INVALID.name(), "Logout Invalid"), HttpStatus.BAD_REQUEST);
+		}
+
+		log.info("Success in user logout. user={}", user.toString());
+
+		subject.logout();
+
+		return new ResponseEntity<>(userMapper.toWS(user.get()), HttpStatus.OK);
+	}
+
+	@PostMapping("/users/{uuid}/forgot_password")
+	public ResponseEntity<?> forgotPassword(@PathVariable String uuid, @Validated @RequestBody UserWS userWS) {
```

Comment:
`forgetPassword` can take just the email address as an input. This assume that the person who can access their email is the actual user.

Reply:
fix